### ISSUE TITLE
Better API documentation

### ISSUE THREAD

**Issue by nicogreenarry** [recharts/recharts#4438]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Much of the API documentation doesn't do much to explain what a given component is and how it would normally be used. Writing a few sentences about what each component is, and how/when you'd use it, could help people understand the library quite a bit better.

For example:
- What is the Brush component? The documentation (https://recharts.org/en-US/api/Brush) just describes the props. I found an example (https://recharts.org/en-US/examples/BrushBarChart), but even the example doesn't make it very clear what Brush is unless you open the CodeSandbox and play around with the code a bit.
- How is the Polygon used? It *sounds* from the name and the props listed in the documentation (https://recharts.org/en-US/api/Polygon) like I can specify points and I'll get a polygon on the chart, but I'm uncertain of that, because I don't see props that I'd expect in that case, e.g. props allowing me to control the style/appearance of the polygon.

### What does the proposed API look like?
n/a

**Comment by ckifer** [comment]
Yep, agreed! The recharts website is hard to maintain because its in another repo.

To help with that we made a storybook (https://master--63da8268a0da9970db6992aa.chromatic.com). That storybook doesn't have a pretty URL, but it at least lives within the recharts repo and is much easier to update.

The storybook should be more up to date than the website is, but still isn't perfect. If anyone wants to help, please feel free to contribute

**Comment by orinamio** [comment]
Hi @ckifer! I'd be happy to help here. Can you assign me to this issue?

**Comment by ckifer** [comment]
Sure! 

This issue is rather large in scope - not sure the best way to structure it, but feel free to create new issues and associate them to this one if you feel the need to do smaller chunks of work at a time (i.e. Improve Brush Documentation, Improve Shape Documentation, etc.)

Another dimension to this is making sure the recharts website is even with the storybook, which it isn't at the moment. Let me know if you need anything!

**Comment by satokenta940** [comment]
I suggest using the advanced documentation tool [Apidog](https://www.apidog.com/?utm_source=&utm_medium=blogger&utm_campaign=test1) to enrich our current API documentation.

**Comment by Kaycee-attah** [comment]
Hi, I'd like to pick this up. It looks like the previous assignee hasn't been active on it. I'm thinking of starting with the Brush component, writing a clear description of what it does and when to use it, and adding a Storybook story with a practical example. Could I be assigned to this?

**Comment by PavelVanecek** [comment]
Go for it. Please prefer to use the website instead of storybooks.

**Comment by Kaycee-attah** [comment]
Okay, noted.

**Comment by Medyooo** [comment]
Hey, is this still being worked on? Happy to pick it up if not :)

**Comment by PavelVanecek** [comment]
Go for it

**Comment by Medyooo** [comment]
Starting with the Polygon component. While testing locally, I noticed `stroke` and `fill` show up in Storybook's generated props table for Polygon, but are missing from the generated website docs (recharts.github.io/en-US/api/Polygon). Might be a gap in how generateApiDoc.ts handles inherited SVG props  will dig into it while I'm in there, but flagging in case it's already known or out of scope for this first pass.

**Comment by PavelVanecek** [comment]
These are inherited from SVG and the website doesn't render that. There are dozens and some are more relevant than others.

**Comment by Medyooo** [comment]
Ah got it, makes sense. I'll mention stroke/fill in the component description text then.
